src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -829,6 +829,46 @@ ov::pass::NopSliceBeforeGatherElements::NopSliceBeforeGatherElements() {
     register_matcher(m, matcher_pass_callback);
 }
 
+ov::pass::PrepareShapeOpsForEliminationAroundBE::PrepareShapeOpsForEliminationAroundBE() {
+    MATCHER_SCOPE(PrepareShapeOpsForEliminationAroundBE);
+    auto first_label = pattern::wrap_type<op::v1::Reshape, op::v0::Squeeze>(pattern::rank_equals(0));
+    auto other_input_label = pattern::any_input(pattern::rank_equals(0));
+    auto binary_op_label = pattern::wrap_type<op::util::BinaryElementwiseArithmetic,
+                                              op::util::BinaryElementwiseComparison,
+                                              op::util::BinaryElementwiseLogical>({first_label, other_input_label},
+                                                                                  pattern::consumers_count(1));
+    auto second_label = pattern::wrap_type<op::v1::Reshape, op::v0::Unsqueeze>({binary_op_label, pattern::any_input()},
+                                                                               pattern::rank_equals(1));
+
+    ov::matcher_pass_callback matcher_pass_callback = [=](pattern::Matcher& m) {
+        const auto& pattern_to_node = m.get_pattern_map();
+
+        auto second_node = pattern_to_node.at(second_label);
+        auto binary = pattern_to_node.at(binary_op_label);
+
+        auto lhs_node =
```

Comment:
can we eliminate the first node (first_label) here?
ov::op::util::clone_try_fold(second_node, {**first_node->input_value(0)**, second_node->input_value(1)});
or we rely on the next elimination transformation?

Reply:
We rely on the following transformations